ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -40,25 +41,28 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
     // boiler-plate
     const {navigation, popup} = useContext(Context);
     const queryParams = new URLSearchParams(location.search);
+    const storage = new ScopedLocalStorage('workflow-details');
 
     const [namespace] = useState(match.params.namespace);
     const [name, setName] = useState(match.params.name);
     const [tab, setTab] = useState(queryParams.get('tab') || 'workflow');
     const [nodeId, setNodeId] = useState(queryParams.get('nodeId'));
+    const [nodePanelView, setNodePanelView] = useState(queryParams.get('nodePanelView') || storage.getItem('nodePanelView', ''));
```

Comment:
Tracks in `queryParams` for instant tab open when sharing a URL that displays said tab, and for browser back-button compatibility.

Tracks in `storage` for consistency across different nodes and workflows.

Priority to `queryParams`

Reply:
we don't use local-storage for other parameters, can we avoid it here?

Reply:
We can if we keep the desired tab in the URL at all times... however this won't persist between different workflows. It will persists between clicking different nodes in a single workflow, which is probably the only desired use case.

Unless you have more thoughts, I'll remove `storage` from this and keep the tab preference in the URL only

Reply:
Done